### PR TITLE
Add "Supporting BFCached Documents" standalone guide

### PR DESCRIPTION
**Pull request by rakina**
See https://github.com/w3ctag/design-reviews/issues/628#issuecomment-1227165134 for context.
cc @cynthia @torgo @smaug---- @annevk @domenic

### CODE REVIEW

**Comment by rakina** [comment]
Post-holiday ping :) I've just updated the README now as requested. The guide can already be read at https://w3ctag.github.io/bfcache-guide/ but I made this PR in case there are some suggestions from TAG/other reviewers.

File: index.bs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+Default Biblio Status: current
+</pre>
+
+<h2 id="intro">Introduction</h2>
+
+Browser implementations may have a back/forward cache, or "BFCache" for short.
+After a user navigates away from a document, the document might be cached in a non-[=Document/fully active=] state,
+and might be reused when the user navigates back.
+In the past, many APIs have missed specifying support for non-fully active documents,
+making them hard to support in various user agents to cache pages in the BFCache,
```
Comment (annevk) [inline index.bs]: that cache pages*?

File: index.bs [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+  APIs that create live connections can pause/close the connection and possibly resume/reopen it later.
+  It's also possible to let the connection stay open to complete existing ongoing requests,
+  and later update the document with the result when it gets restored, if appropriate (e.g.
+  resource loads).
+</div>
+<div class="example">
+  APIs that hold non-exclusive resources
+  may be able to release the resource when the document becomes not fully active,
+  and re-acquire them when it becomes [=Document/fully active=] again
+  (Screen Wake Lock API is already <a href="https://w3c.github.io/screen-wake-lock/#handling-document-loss-of-full-activity">doing</a> the first part).
```
Comment (annevk) [inline index.bs:114]: This is pretty poor example in that there's no explicit hook tying this to some algorithm that runs when the fully active state changes. Are there hooks for that?

**Review by annevk — COMMENTED** [review]
I'm missing some guidance around "in parallel".

At least abstractly it seems that whenever you do "in parallel" followed by "queue a global task" there's an opportunity for a document to become inactive.

However, HTML will block task processing in that case until the document is active again so you can't really know.

I'm also vaguely reminded of an issue around promises here.

E.g., if we consider https://storage.spec.whatwg.org/#dom-storagemanager-persisted. Should that just work as-is or will any changes be needed?

**Comment by annevk** [comment]
(Also, thanks for pushing this! It's great that we give this some more attention.)

File: index.bs (hunk 2 above)
Comment (rakina) [reply index.bs:114]: Unfortunately we don't have hooks for "fully active" state changes. Maybe we should add that? Do you know what's the best way to do that?

File: index.bs (hunk 1 above)
Comment (rakina) [reply index.bs]: Done, thanks!

**Comment by rakina** [comment]
Thanks for the review!

> I'm missing some guidance around "in parallel".
> 
> At least abstractly it seems that whenever you do "in parallel" followed by "queue a global task" there's an opportunity for a document to become inactive.
> 
> However, HTML will block task processing in that case until the document is active again so you can't really know.
> 
> I'm also vaguely reminded of an issue around promises here.
> 
> E.g., if we consider https://storage.spec.whatwg.org/#dom-storagemanager-persisted. Should that just work as-is or will any changes be needed?

Yes, I think tasks that are posted might not actually be run until the document becomes active again (or if the document gets evicted, never run at all). Maybe we can add a section that mentions this explicitly just so that people are aware that this is possible? They will not run when the document is inactive, so hopefully there's no problems with things thinking it's an inactive document in a posted task? (I'm not really sure what the problem might be in your example)

File: index.bs (hunk 2 above)
Comment (annevk) [reply index.bs:114]: I suspect we want something similar to HTML's "update the rendering" where specifications have an algorithm that HTML ends up invoking when it changes the state.

**Comment by annevk** [comment]
Maybe the better example would be the `persist()` method in the same document. It's rather unclear whether or not the state ends up changing or the user gets prompted (or is still prompted when going back if they had not answered yet) depending on when the fully active transition happens. I mean, there is a certainly a way to read it that answers these questions, kinda, but I doubt it's accurate.

File: index.bs (hunk 2 above)
Comment (atanassov) [reply index.bs:114]: These are all valid points though not necessarily strong enough to hold back the forking of this document. My suggestion is to open separate issues against the newly forked BFCache document.

Comment (annevk) [reply index.bs:114]: I find that a bit rude. I was asked for feedback here and only now you're telling me I should have been filing issues instead.

Comment (rakina) [reply index.bs:114]: Hi @atanassov, I think the merge of this PR wasn't urgently needed. The HTML file of the bikeshed file is already part of the gh-pages branch so the document was already accessible from https://w3ctag.github.io/bfcache-guide/. I had intended to merge the reviewed version of the document to that branch after all the issues are resolved and this got LGTMs from other browsers.

@annevk, I'm very sorry for the delay in addressing the reviews here and the sudden merge, I assure you that I will address the issues and update the guide. I have filed [an issue](https://github.com/w3ctag/bfcache-guide/issues/2) for the "in parallel" one, and [another issue](https://github.com/whatwg/html/issues/8872) for the "fully active" state change listener. Thanks a lot for reviewing this PR and all the help!!

**Pull request merged**